rockchip: make NIC name predictable for Radxa E52C/ROCK 5 ITX/ROCK 5T

RadxaNaoki opened the pull request:
The probe order for PCIe buses and devices is non-deterministic, making the names eth0 and eth1 unpredictable (they may be swapped).

This patch fixes the names by referencing the device path using `ucidef_set_network_device_path`.

The mapping between silkscreen labels on the board/case and OpenWrt interface names is as follows:

- E52C
  LAN: lan
  WAN: wan

- ROCK 5 ITX
  RJ45 1: eth0
  RJ45 2: eth1

- ROCK 5T
  RJ45_1: eth0
  RJ45_2: eth1

For Radxa E52C, this breaks compatibility of the network config; therefore, set DEVICE_COMPAT_VERSION to `1.1`.

Fixes: d16d2765bd ("rockchip: add support for Radxa E52C")
Fixes: 0839345211 ("rockchip: add support for Radxa ROCK 5 ITX/ITX+")
Fixes: 4a78af9876 ("rockchip: add support for Radxa ROCK 5T")
Link: https://github.com/openwrt/openwrt/issues/20202

1715173329 commented on target/linux/rockchip/armv8/base-files/etc/board.d/02_network:
```diff
@@ -37,6 +34,21 @@ rockchip_setup_interfaces()
 	friendlyarm,nanopi-r6s)
 		ucidef_set_interfaces_lan_wan 'eth0 eth2' 'eth1'
 		;;
+	radxa,e52c)
```
Since you introduced breaking changes to the network config, please add device compat version.

```
define Device/radxa_e52c
  $(Device/rk3582)
  DEVICE_VENDOR := Radxa
  DEVICE_MODEL := E52C
  DEVICE_DTS := rk3582-radxa-e52c
  UBOOT_DEVICE_NAME := generic-rk3588
  DEVICE_PACKAGES := blkdiscard kmod-r8169
  DEVICE_COMPAT_VERSION := 1.1
  DEVICE_COMPAT_MESSAGE := Network interface names have been changed
endef
```

1715173329 replied:
> Ah, well, is it required for E52C, even though it's not in stable release yet?

Yes, please.

> I understand there is also the option of keeping the interface names as eth0/eth1 instead of using lan/wan, but it's difficult to decide which is better. Would aligning this with the case printing be a better approach?

We do not have explicit rules for this - at least for now.
I'm personally okay for both.

RadxaNaoki replied:
DEVICE_COMPAT_VERSION and DEVICE_COMPAT_MESSAGE are added.
Thank you for the review!

1715173329 replied:
Please add script to set correct compat version for this device, fyi: [05_compat-version](https://github.com/immortalwrt/immortalwrt/blob/df83fe19b3ca29e2091f10643b728d83d8746ca1/target/linux/rockchip/armv8/base-files/etc/board.d/05_compat-version).
The compat version defaults to 1.0 so each time users perform sysupgrade the utility will throw a warning.

RadxaNaoki replied:
https://github.com/openwrt/openwrt/pull/21533

RadxaNaoki committed:
rockchip: make NIC name predictable for Radxa E52C/ROCK 5 ITX/ROCK 5T

The probe order for PCIe buses and devices is non-deterministic,
making the names eth0 and eth1 unpredictable (they may be swapped).

This patch fixes the names by referencing the device path using
`ucidef_set_network_device_path`.

The mapping between silkscreen labels on the board/case and OpenWrt
interface names is as follows:

- E52C
  LAN: lan
  WAN: wan

- ROCK 5 ITX
  RJ45 1: eth0
  RJ45 2: eth1

- ROCK 5T
  RJ45_1: eth0
  RJ45_2: eth1

For Radxa E52C, this breaks compatibility of the network config;
therefore, set DEVICE_COMPAT_VERSION to `1.1`.

Fixes: d16d2765bd ("rockchip: add support for Radxa E52C")
Fixes: 0839345211 ("rockchip: add support for Radxa ROCK 5 ITX/ITX+")
Fixes: 4a78af9876 ("rockchip: add support for Radxa ROCK 5T")
Link: https://github.com/openwrt/openwrt/issues/20202
Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/20608
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki commented:
> There's no breaking change to these boards so not needed.

Well, I think it will be necessary to replace board.json to **reflect** this change, and that this will require `sysupgrade -n`.

1715173329 commented:
You can mention this in commit message, or add an uci-defaults script to apply these changes _if necessary_.

RadxaNaoki commented:
> Has anyone encountered LAN/WAN swap issues on E52C? If not, I think the E52C board doesn't need this change.

So far, I do not believe I have observed this issue on the E52C. However, because I believe the Network Interface Name is unpredictable (at least on the rk3588) and that `ucidef_set_network_device_path` is useful, I think that `ucidef_set_network_device_path` should be used when there are multiple NICs.

RadxaNaoki commented:
> You can mention this in commit message, or add an uci-defaults script to apply these changes _if necessary_.

I confirmed that board.json is not backed up by running just `sysupgrade openwrt...sysupgrade.img.gz`. So the current patch is fine.

RadxaNaoki commented:
> It seems that most RK3588 devices do not have this problem on OpenWRT. I think it's better not to change the configuration if there's no problem. (At least the E52C does not need a fixes tag.)

Do "most RK3588 devices" have PCIe NIC?

> By the way, have you tried using openwrt,netdev-name?

Where should I use it specifically?
